crates/web-plugins/didcomm-messaging/src/did_rotation/did_rotation.rs

```diff
@@ -0,0 +1,350 @@
+use super::errors::RotationError;
+use crate::{didcomm::bridge::LocalDIDResolver, model::stateful::entity::Connection};
+use axum::response::{IntoResponse, Response};
+use database::Repository;
+use didcomm::{FromPrior, Message};
+use mongodb::bson::doc;
+use std::sync::Arc;
+
+/// https://identity.foundation/didcomm-messaging/spec/#did-rotation
+pub async fn did_rotation(
+    msg: Message,
+    connection_repos: &Arc<dyn Repository<Connection>>,
+) -> Result<(), Response> {
+    // Check if from_prior is none
+    if msg.from_prior.is_none() {
+    } else {
```

Comment:
This is what I meant by early return. You don't need an else branch with this logic.
```suggestion
    if msg.from_prior.is_none() {
        return Ok(());
    }
```

Reply:
:+1: